src/dfcx_scrapi/tools/search_util.py

```diff
@@ -497,6 +533,360 @@ def find_list_parameters(self, agent_id):
+        elif isinstance(cxobj, DFCXEventHandler):
+            fulfillment_type = "trigger_fulfillment"
+            location = "Event Handler"
+            if page_param:
+                location = "Form Filling Event Handler"
+        elif isinstance(cxobj, DFCXPage):
+            fulfillment_type = "entry_fulfillment"
+            location = "Entry Fulfillment"
+        else:
+            fulfillment_type = "initial_prompt_fulfillment"
+            location = "Parameter Filling"
+
+        # Error checking
+        if hasattr(cxobj,fulfillment_type) and getattr(cxobj,fulfillment_type):
+            ful_data = getattr(cxobj, fulfillment_type)
+            spa = "set_parameter_actions"
+            if hasattr(ful_data, spa) and ful_data.set_parameter_actions:
+                for param_data in ful_data.set_parameter_actions:
+                    param_name = param_data.parameter
+                    param_value = param_data.value
+                    if isinstance(param_value, MapComposite):
+                        param_value=self.convert_protobuf(param_value)
+                    if isinstance(param_value, RepeatedComposite):
+                        param_value=self.convert_protobuf(param_value)
+                    flow_names.append(flow_name)
+                    page_names.append(page_name)
+                    if hasattr(cxobj, "intent") and cxobj.intent:
+                        if cxobj.intent in self.intents_map:
+                            route_intents.append(
+                                self.intents_map[cxobj.intent]
+                            )
+                        else:
+                            route_intents.append("")
+                    else:
+                        route_intents.append("")
+                    if hasattr(cxobj, "event") and cxobj.event:
+                        route_events.append(cxobj.event)
+                    else:
+                        route_events.append("")
+                    route_group_names.append(route_group)
+                    page_param_names.append(page_param)
+                    if hasattr(cxobj,"condition") and cxobj.condition:
+                        route_conditions.append(cxobj.condition)
+                    else:
+                        route_conditions.append("")
+                    location_names.append(location)
+                    parameter_preset_names.append(param_name)
+                    parameter_preset_values.append(param_value)
+
+        return pd.DataFrame({"flow": flow_names,
+            "page": page_names,
+            "intent": route_intents,
+            "event": route_events,
+            "route_group": route_group_names,
+            "condition": route_conditions,
+            "form_filling": page_param_names,
+            "location": location_names,
+            "param_name": parameter_preset_names,
+            "param_value": parameter_preset_values
+        })
+
+    def convert_protobuf(self, obj):
```

Comment:
@SeanScripts @skyyeat this code is redundant and can be removed.

There is already a set of methods in `scrapi_base.ScrapiBase` that handle recursion of `MapComposite` and `RepeatedComposite` types.

See:
* [recurse_proto_repeated_composite](https://github.com/GoogleCloudPlatform/dfcx-scrapi/blob/main/src/dfcx_scrapi/core/scrapi_base.py#L245)
* [recurse_proto_marshall_to_dict](https://github.com/GoogleCloudPlatform/dfcx-scrapi/blob/main/src/dfcx_scrapi/core/scrapi_base.py#L259)
* [cx_object_to_dict](https://github.com/GoogleCloudPlatform/dfcx-scrapi/blob/main/src/dfcx_scrapi/core/scrapi_base.py#L116)

If for some reason these don't fit your exact use case, please take one of the following actions:
- add a new method to scrapi_base to be used by all classes
- modify one of the existing methods in scrapi_base appropriately

If you choose the latter, ensure that you've tested the other classes thoroughly, as there could be many, many dependencies on these methods throughout.